rfcs/2023-rsp-testing-utils.md

```diff
@@ -0,0 +1,113 @@
+
+These util classes could live in the `@react-aria/test-util` package and re-exported from `react-aria-components`.
+
+## Documentation
+
+Documentation for the first two parts described above should go in the existing "Testing" page that we have [in the docs](https://react-spectrum.adobe.com/react-spectrum/testing.html#testing) already. We'll want to consider having a similar page in the React Aria Component/React Aria docs sections as well or moving that page up in hierarchy so it lives on the [main page](https://react-spectrum.adobe.com/index.html). The test util classes could also be documented in the same page as the rest of the testing info. I think we should also include a testing section in each of the individual component pages that highlights what test util class can be used (aka mapping Picker to SelectTester), documents the class's methods and options, and links back to the overall testing FAQ for common issues/gotchas that are related to said component. The main testing page should also show a high level example of using a test util class in a example test.
+
+## Drawbacks
+
+My primary concern is whether or not some of the mocks truly work in other test suite configurations. Historically, we've seen other teams struggle with a small subset of the mocks not resolving their test rendering issues (specifically with ListBox and the `clientHeight`). Making sure the interaction utils aren't brittle may be a challenge since the way we test in RSP is with components in isolation versus other teams testing entire app flows that have many elements rendered at once. For instance, we can easily find the ListBox tied to a Select element by querying for `role="listbox"` in our RSP test, but doing so in a page filled with other ListBoxes would involve looking up the `id` from the Select's `aria-controls`.
+
+
+## Backwards Compatibility Analysis
+
+This is a backwards compatible change since no existing test utils are exposed from our library today.
+
+## Alternatives
+
+Unknown, haven't done research here yet.
+
+## Open Questions
+
+* What should be exposed as a test util vs documented? Since it will take a longer amount of time required to spin up the test util classes, should we expose the interaction util functions we already have even though they will be replaced by the test util classes?
+* Where should these utils live? Exported from the monopackage as well? Or separated into its own test-util package? Re-exported from existing packages (react-aria-components, react-aria, react-spectrum)?
```

Comment:
Exporting from the existing monopackages seems fine, unless we're concerned with crowding the namespace.

Reply:
The namespace is really my main concern, might make IDE autocomplete too busy

Reply:
Yeah I would not mix test and production code as I mentioned above. I think importing from the separate package in tests is totally fine.